src/ray/core_worker/task_interface.cc

```diff
@@ -1,26 +1,155 @@
+
+  auto num_returns = call_options.num_returns;
+  (*return_ids).resize(num_returns);
+  for (int i = 0; i < num_returns; i++) {
+    (*return_ids)[i] = ObjectID::ForTaskReturn(task_id, i + 1);
+  }
+
+  std::vector<std::shared_ptr<raylet::TaskArgument>> task_arguments;
+  for (auto &arg : args) {
+    if (arg.IsPassedByReference()) {
+      std::vector<ObjectID> references{arg.GetReference()};
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByReference>(references));
+    } else {
+      auto data = arg.GetValue();
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByValue>(data->Data(), data->Size()));
+    }
+  }
+
+  ::Language language = (function.language == ray::WorkerLanguage::JAVA)
+                            ? (::Language::JAVA)
+                            : (::Language::PYTHON);
+
+  ray::raylet::TaskSpecification spec(context.GetCurrentDriverID(),
+                                      context.GetCurrentTaskID(), next_task_index,
+                                      task_arguments, num_returns, call_options.resources,
+                                      language, function.function_descriptor);
+
+  std::vector<ObjectID> execution_dependencies;
+  return core_worker_.raylet_client_->SubmitTask(execution_dependencies, spec);
 }
 
 Status CoreWorkerTaskInterface::CreateActor(
     const RayFunction &function, const std::vector<TaskArg> &args,
-    const ActorCreationOptions &actor_creation_options, ActorHandle *actor_handle) {
-  return Status::OK();
+    const ActorCreationOptions &actor_creation_options,
+    std::unique_ptr<ActorHandle> *actor_handle) {
+  auto &context = core_worker_.worker_context_;
+  auto next_task_index = context.GetNextTaskIndex();
+  const auto task_id = GenerateTaskId(context.GetCurrentDriverID(),
+                                      context.GetCurrentTaskID(), next_task_index);
+
+  std::vector<ObjectID> return_ids;
+  return_ids.push_back(ObjectID::ForTaskReturn(task_id, 1));
+  ActorID actor_creation_id = ActorID::FromBinary(return_ids[0].Binary());
+
+  *actor_handle = std::unique_ptr<ActorHandle>(
+      new ActorHandle(actor_creation_id, ActorHandleID::Nil()));
+  (*actor_handle)->IncreaseTaskCounter();
+  (*actor_handle)->SetActorCursor(return_ids[0]);
+
+  std::vector<std::shared_ptr<raylet::TaskArgument>> task_arguments;
+  for (auto &arg : args) {
+    if (arg.IsPassedByReference()) {
+      std::vector<ObjectID> references{arg.GetReference()};
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByReference>(references));
+    } else {
+      auto data = arg.GetValue();
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByValue>(data->Data(), data->Size()));
+    }
+  }
+
+  // Note that the caller is supposed to specify required placement resources
+  // correctly via actor_creation_options.resources.
```

Comment:
it's a bit weird that this comment appears here. because it's not related to the following code

Reply:
updated. moved this down to the top of creating the task spec.